recipes/imagemagick/all/conanfile.py

```diff
@@ -0,0 +1,518 @@
+    def _is_msvc(self):
+        return self.settings.compiler == "Visual Studio"
+
+    @property
+    def _modules(self):
+        return ["Magick++", "MagickWand", "MagickCore"]
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+            raise ConanInvalidConfiguration(
+                "Windows builds of ImageMagick require MFC which cannot currently be sourced from CCI."
+            )
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.options.with_zlib:
+            self.requires("zlib/1.2.11")
+        if self.options.with_bzlib:
+            self.requires("bzip2/1.0.8")
+        if self.options.with_lzma:
+            self.requires("xz_utils/5.2.5")
+        if self.options.with_lcms:
+            self.requires("lcms/2.11")
+        if self.options.with_openexr:
+            self.requires("openexr/2.5.7")
+        if self.options.with_jpeg == "libjpeg":
+            self.requires("libjpeg/9d")
+        elif self.options.with_jpeg == "libjpeg-turbo":
+            self.requires("libjpeg-turbo/2.1.0")
+        if self.options.with_openjp2:
+            self.requires("openjpeg/2.4.0")
+        if self.options.with_pango:
+            self.requires("pango/1.48.5")
+        if self.options.with_png:
+            self.requires("libpng/1.6.37")
+        if self.options.with_tiff:
+            self.requires("libtiff/4.3.0")
+        if self.options.with_webp:
+            self.requires("libwebp/1.2.0")
+        if self.options.with_xml2:
+            self.requires("libxml2/2.9.10")
+        if self.options.with_freetype:
+            self.requires("freetype/2.10.4")
+        # TODO add when available
+        if self.options.with_djvu:
+            # FIXME: missing djvu recipe
+            self.output.warn(
+                "There is no djvu package available on Conan (yet). This recipe will use the one present on the system (if available)."
+            )
+
+    def source(self):
+        tools.get(
+            **self.conan_data["sources"][self.version]["source"],
+            destination=self._source_subfolder,
+            strip_root=True
+        )
+
+        visualmagick_version = list(
+            self.conan_data["sources"][self.version]["visualmagick"].keys()
+        )[0]
+        tools.get(
+            **self.conan_data["sources"][self.version]["visualmagick"][
+                visualmagick_version
+            ],
+            destination="VisualMagick",
+            strip_root=True
+        )
+
+    def build(self):
+        if self._is_msvc:
+            self._build_msvc()
+        else:
+            with tools.chdir(self._source_subfolder):
+                env_build = self._build_configure()
+                env_build.make()
+
+    def _build_msvc(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, {}):
+            tools.patch(**patch)
```

Comment:
The only patch is "remove deps" should it not be applied on all systems?

Reply:
I'm not sure this is from the original bincrafters recipe, I haven't changed anything related to windows builds since they don't currently work.